sdk/extensions/Microsoft.Extensions.Azure.WebJobs/tests/shared/TestHelpers.cs

```diff
@@ -0,0 +1,324 @@
+                        services.AddSingleton<INameResolver>(nameResolver);
+                    }
+
+                    if (activator != null)
+                    {
+                        services.AddSingleton<IJobActivator>(activator);
+                    }
+                });
+        }
+
+        public static IHostBuilder ConfigureTestLogger(this IHostBuilder builder)
+        {
+            return builder.ConfigureLogging(logging =>
+             {
+                 logging.AddProvider(new TestLoggerProvider());
+             });
+        }
+
+
+        public static IHostBuilder ConfigureTypeLocator(this IHostBuilder builder, params Type[] types)
+        {
+            return builder.ConfigureServices(services =>
+            {
+                services.AddSingleton<ITypeLocator>(new FakeTypeLocator(types));
+            });
+        }
+
+        public static TestLoggerProvider GetTestLoggerProvider(this IHost host)
+        {
+            return host.Services.GetServices<ILoggerProvider>().OfType<TestLoggerProvider>().Single();
+        }
+
+        public static TExtension GetExtension<TExtension>(this IHost host)
+        {
+            return host.Services.GetServices<IExtensionConfigProvider>().OfType<TExtension>().SingleOrDefault();
+        }
+
+        public static JobHost GetJobHost(this IHost host)
+        {
+            return host.Services.GetService<IJobHost>() as JobHost;
+        }
+
+        public static JobHost<TProgram> GetJobHost<TProgram>(this IHost host)
+        {
+            return host.Services.GetService<IJobHost>() as JobHost<TProgram>;
+        }
+
+        public static async Task CallAsync<T>(this JobHost host, string methodName, object arguments)
+        {
+            await host.CallAsync(typeof(T).GetMethod(methodName), arguments);
+        }
+
+        public static async Task CallAsync<T>(this JobHost host, string methodName)
+        {
+            await host.CallAsync(typeof(T).GetMethod(methodName));
+        }
+
+        public static TOptions GetOptions<TOptions>(this IHost host) where TOptions : class, new()
+        {
+            return host.Services.GetService<IOptions<TOptions>>().Value;
+        }
+
+#pragma warning disable CS0618 // Type or member is obsolete
+        public static IJobHostMetadataProvider CreateMetadataProvider(this IHost host)
+        {
+            return host.Services.GetService<IJobHostMetadataProvider>();
+#pragma warning restore CS0618 // Type or member is obsolete
+        }
+
+        public static List<string> GetAssemblyReferences(Assembly assembly)
+        {
+            var assemblyRefs = assembly.GetReferencedAssemblies();
+            var names = (from assemblyRef in assemblyRefs
+                         orderby assemblyRef.Name.ToLowerInvariant()
+                         select assemblyRef.Name).ToList();
+            return names;
+        }
+
+        public static void AssertPublicTypes(IEnumerable<string> expected, Assembly assembly)
```

Comment:
NIT: Do we actually need these?  They seem to be unused in this PR.

Reply:
I copied them bulk to simplify migration for other extension tests and not force them to add methods one-by-one. Plan to clean up when we get a few merged and figure out what's used and what's not.

Reply:
SGTM!